### PR TITLE
Use batch processing to process timeline and cpu profile data.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This keeps the UI at 60 FPS even when we are doing heavy computation. Previously, this work was blocking the UI thread, so processing it in batches by awaiting a small delay allows the UI to stay responsive.

Added a linear progress indicator to inform the user of the loading progression:

Timeline data:
processingTimelineLoadingBar

Cpu profile data:
processingLoadingBar

As a bonus, this fix translated to the dart:html app, too. Fixes https://github.com/flutter/devtools/issues/1551.

Screen Shot 2020-01-27 at 9 28 52 AM

### CODE REVIEW

File: packages/devtools_app/lib/src/flutter/common_widgets.dart [hunk 1]
```diff
@@ -268,17 +272,54 @@ Widget recordingInfo({
...
       const CircularProgressIndicator(),
     ],
   );
+  final processingStatus = processingInfo(
+    key: processingStatusKey,
+    progressValue: progressValue,
+    processedObject: recordedObject,
+  );
+
+  Widget child;
```
Comment (jacob314) [inline packages/devtools_app/lib/src/flutter/common_widgets.dart]: nit: make this a helper method so these cases can all be return statements.

File: packages/devtools_app/lib/src/profiler/cpu_profile_transformer.dart [hunk 2]
```diff
@@ -29,20 +96,9 @@ class CpuProfileTransformer {
...
-// https://github.com/flutter/devtools/issues/1529.
-//    assert(
-//      cpuProfileData.profileMetaData.sampleCount ==
-//          cpuProfileData.cpuProfileRoot.inclusiveSampleCount,
-//      'SampleCount from response (${cpuProfileData.profileMetaData.sampleCount})'
-//      ' != sample count from root '
-//      '(${cpuProfileData.cpuProfileRoot.inclusiveSampleCount})',
-//    );
+    }
+    _stackFramesProcessed +=
```
Comment (jacob314) [inline packages/devtools_app/lib/src/profiler/cpu_profile_transformer.dart]: nit: this is correct but might as well just increment
_stackFramesProcessed within the for loop.
The performance hit won't matter and that way there can't be bugs if for example an exception is thrown processing a frame or if the logic is tweaked.

File: packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart [hunk 3]
```diff
@@ -79,9 +83,23 @@ class TimelineScreenBodyState extends State<TimelineScreenBody>
 
     cancel();
     addAutoDisposeListener(controller.timelineModeNotifier);
-    autoDispose(controller.fullTimeline.onTimelineProcessed.listen((_) {
-      setState(() {});
-    }));
+    addAutoDisposeListener(controller.fullTimeline.recordingNotifier, () {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart]: this logic is duplicated. maybe add a helper widget that supports using these value notifiers to decide the loading status to show and leaving the other code using ValueListenableBuilder as is?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart]: Added a todo to do this in a follow up CL

**Comment by jacob314** [comment]
lgtm_dog2

**Pull request merged**